lib/percy/network_helpers.rb

```diff
@@ -1,16 +1,32 @@
 require 'socket'
 require 'excon'
+require 'timeout'
 
 module Percy
   class NetworkHelpers
+    MIN_PORT = 1_024 # 0-1023 are system reserved
+    MAX_PORT = 65_535 # (2^16) -1
+
     class ServerDown < RuntimeError; end
 
     def self.random_open_port
-      # Using a port of "0" relies on the system to pick an open port.
-      server = TCPServer.new('127.0.0.1', 0)
-      port = server.addr[1]
-      server.close
-      port
+      loop do
```

Comment:
Per discussion, maybe change this to something like `MAX_PORT_ATTEMPTS.times do` with a raise outside of it, so at least it doesn't hang forever.